libr/bin/dwarf.c

```diff
@@ -480,8 +480,11 @@ static const ut8 *parse_line_header_source(RBinFile *bf, const ut8 *buf, const u
 					hdr->file_names[count].mod_time = mod_time;
 					hdr->file_names[count].file_len = file_len;
 				}
-				free (comp_dir);
-				free (pinclude_dir);
+				if (comp_dir) {
```

Comment:
can't you just do `R_FREE(include_dir); R_FREE(comp_dir);` without checking for comp_dir? If they are null, it's going to be a no-op anyway.

Reply:
I can't, because based on the code, **include_dir** and **pinclude_dir** point to the same thing in case **comp_dir** isn't NULL, that's why I check if it is NULL, if it's not, I free **include_dir** aswell, its to prevent double free in the case they both point to the same thing.

Reply:
:+1: